core/src/main/java/org/apache/accumulo/core/data/RangedObject.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.data;
+
+/**
+ * Marker interface that declares the object is a type that represents a range
+ */
+public interface RangedObject<T> {
```

Comment:
I'm not sure this interface makes sense to add. Although it's helpful to have a similar method on the `Range` class and `KeyExtent` because they serve similar roles in our public API and internal implementation, respectively, they are not closely related enough to require an interface to try to make them related. The need to use a generic type for its argument reveals how distantly related they are, as well as the fact that they have independent implementations, and we never assign either to any variable of the interface type for any purpose.

I think it is sufficient to add the `overlaps` method to `Range` (with a corresponding `@since` tag on the new method) without adding this interface to the public API.

Reply:
Let's wait to include it alongside that proposed change, if/when it manifests. I think it's premature here, as its API needs might be slightly different than what you have here by the time it is ready.

Reply:
Removed RangedObject in 5b98e2a